### PR TITLE
Added QA readme

### PR DESCRIPTION
**Pull request by Comoque1**
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [x] Documentation is added
- [ ] Test cases are added or updated
- [ ] I've read the contributing document https://github.com/cherrytwist/.github/blob/master/CONTRIBUTING.md
- [ ] I've read and understand the Code of Conduct https://github.com/cherrytwist/.github/blob/master/CODE_OF_CONDUCT.md
- [ ] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/cherrytwist/Coordination/blob/master/LICENSE 
     and the contributor license agreement: tba

### CODE REVIEW

File: QA/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    - GraphQL Exploratory testing
+        
+- ### Non-functional testing:
+    - [Performance testing](https://github.com/cherrytwist/Server/tree/develop/test/performance)
+    - Security testing<sup>*</sup>
+    - Database migration<sup>*</sup>
+
+- ### Static testing:
+    - Code review
+    - Static code analyses tools<sup>*</sup>
```
Comment (techsmyth) [inline QA/README.md]: We do have BetterCodeHub integrated in both server + client

File: QA/README.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ 
+## "Client.Web" - testing types and levels:
```
Comment (techsmyth) [inline QA/README.md:41]: Please put in a comment that the focus for QA is on the server - client QA to be ramped up

File: QA/README.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+This document provides overview of "Cherrytwist" quality assurance and testing activities. Some of them are in progress and some are to be started.
+
+# Process
+ 
+- Perform functional/non-functional testing types on reviewed "github" issues
+    - Depending on the "github" issue scope, testing is performed on different environments 
+- Log defects, communicate  and retest them
+- Perform regression testing as part of "github" issues testing
+- Automate/maintain test cases on different levels
+- Test data management
```
Comment (techsmyth) [inline QA/README.md:12]: Package dependency audit: regular review of reported vulnerabilities in packages dependent on and then address

File: QA/README.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+
+# Testing types and levels
+
+"Cherrytwist" consists of multiple services. The following test types are being applied among the different test levels:
+
+## "Server" - testing types and levels:
+- ### Functional testing: 
+    - Unit testing<sup>*</sup>
```
Comment (techsmyth) [inline QA/README.md:25]: Do we not have any unit tests at the moment?

File: QA/README.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    - [GraphQL API integration tests](https://github.com/cherrytwist/Server/tree/develop/test)
+    - GraphQL Exploratory testing
+        
+- ### Non-functional testing:
+    - [Performance testing](https://github.com/cherrytwist/Server/tree/develop/test/performance)
+    - Security testing<sup>*</sup>
+    - Database migration<sup>*</sup>
```
Comment (techsmyth) [inline QA/README.md]: This you do have - cannot be automated as each one will be different but there is a process for the migrations. Might be worth referring to that document?

File: QA/README.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+    - Number of defects<sup>*</sup>
+    - Number of production environment defects by severity<sup>*</sup>
+    - Number of development environment defects by severity<sup>*</sup>
+
+# Environments
+- Local - setup for [Server](https://github.com/cherrytwist/Server) and [Client.Web](https://github.com/cherrytwist/Client.Web)
+- [Development](https://dev.cherrytwist.org/) - auto-deployed after each merge to development branch
+- [Test](https://test.cherrytwist.org/) - QA deploys from "github" repositories action, the required branch
+- [Acc](https://acc.odyssey.ninja:3000/ecoverse) - Project manager deploys from "github" repositories action, from master branch
```
Comment (techsmyth) [inline QA/README.md]: Should not refer to Odyssey environments there - they are not part of Cherrytwist but our first client

File: QA/README.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+- Defects<sup>*</sup>
+
+    - Number of defects<sup>*</sup>
+    - Number of production environment defects by severity<sup>*</sup>
+    - Number of development environment defects by severity<sup>*</sup>
+
+# Environments
+- Local - setup for [Server](https://github.com/cherrytwist/Server) and [Client.Web](https://github.com/cherrytwist/Client.Web)
+- [Development](https://dev.cherrytwist.org/) - auto-deployed after each merge to development branch
+- [Test](https://test.cherrytwist.org/) - QA deploys from "github" repositories action, the required branch
```
Comment (techsmyth) [inline QA/README.md]: Deployed from which branch?

File: QA/README.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+- Defects<sup>*</sup>
+
+    - Number of defects<sup>*</sup>
+    - Number of production environment defects by severity<sup>*</sup>
+    - Number of development environment defects by severity<sup>*</sup>
+
+# Environments
+- Local - setup for [Server](https://github.com/cherrytwist/Server) and [Client.Web](https://github.com/cherrytwist/Client.Web)
+- [Development](https://dev.cherrytwist.org/) - auto-deployed after each merge to development branch
```
Comment (techsmyth) [inline QA/README.md]: "develop" branch

File: QA/README.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+- [Prod](https://prod.odyssey.ninja:3000/ecoverse) - Project manager deploys from "github" repositories action, from master branch
+
+# Tools
+- [Jest](https://jestjs.io/) - javascript automation test framework
+- [Jmeter](https://jmeter.apache.org/) - performance test tool 
+- [Postman](https://www.postman.com/) - API testing tool
+- [MySQL 8](https://dev.mysql.com/doc/relnotes/mysql/8.0/en/) - query data and testing the systems integration
+- [Populator.GSheet](https://github.com/cherrytwist/Populator.GSheet) - test data population
+- Test management system - for the moment, the manual testing is tracked within "github" issues
+    - [KIWI TCMS](https://kiwitcms.org/features/) is being evaluated for manual test case management
```
Comment (techsmyth) [inline QA/README.md:90]: BetterCodeHub should also be listed here, with their logo

**Review by techsmyth — COMMENTED** [review]
Added a lot of comments, but that should not take away from this being a very useful document to have in place - thanks!

File: QA/README.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
+# CherryTwist - Quality Assurance
+
+This document provides overview of "Cherrytwist" quality assurance and testing activities. Some of them are in progress and some are to be started.
```
Comment (techsmyth) [inline QA/README.md]: This document provides an overview of the quality assurance and testing activities that are carried out on the Cherrytwist platform
- basically do not like Cherrytwist in quotes.

File: QA/README.md [hunk 11]
```diff
@@ -58,12 +74,13 @@ This document provides overview of "Cherrytwist" quality assurance and testing a
 
 # Environments
 - Local - setup for [Server](https://github.com/cherrytwist/Server) and [Client.Web](https://github.com/cherrytwist/Client.Web)
-- [Development](https://dev.cherrytwist.org/) - auto-deployed after each merge to development branch
-- [Test](https://test.cherrytwist.org/) - QA deploys from "github" repositories action, the required branch
-- [Acc](https://acc.odyssey.ninja:3000/ecoverse) - Project manager deploys from "github" repositories action, from master branch 
-- [Prod](https://prod.odyssey.ninja:3000/ecoverse) - Project manager deploys from "github" repositories action, from master branch
+- [Development](https://dev.cherrytwist.org/) - auto-deployed after each merge to develop branch
+- [Test](https://test.cherrytwist.org/) - QA deploys from "github" repositories action, the branch version that is being tested
+- [Acc](https://acc.odyssey.ninja:3000/) - Project manager deploys from "github" repositories action, from master branch
```
Comment (techsmyth) [inline QA/README.md]: We should not be listing the Odyssey Acc or Prod enviornments on our Cherrytwist repo; they are a partner install

File: QA/README.md (hunk 5 above)
Comment (Comoque1) [reply QA/README.md]: Added  dbmigration test document, to the server repository. Will update the link, once the PR for this document is approved and merged with develop branch.

File: QA/README.md (hunk 4 above)
Comment (Comoque1) [reply QA/README.md:25]: The existing ones and autogenerated and I have to fix them. Once we have reasonable unit test suites for the different entities, I'll update the document.

**Review by techsmyth — APPROVED** [review]

**Pull request merged**